pkgpanda/__init__.py

```diff
@@ -792,10 +797,10 @@ def _get_service_names(_dir):
             if package.username is not None:
                 sysusers.add_user(package.username, package.group)
 
-            # Ensure the state directory in `/var/lib/dcos` exists
+            # Ensure the state directory exists
             # TODO(cmaloney): On upgrade take a snapshot?
             if self.__manage_state_dir:
-                state_dir_path = '/var/lib/dcos/{}'.format(package.name)
+                state_dir_path = self.__state_dir_root + '/' + package.name
```

Comment:
[not a review comment] 

I can understand when when we use  + '/' + for doing path constructions.
When using os.path.join, we have to be careful with the path join protocols, like 
```
>>> r = '/root'
>>> a = '/a'
>>> b = 'b'
>>> os.path.join(r, a)
'/a'
>>> os.path.join(r, b)
'/root/b'
```

* By doing '/' +, we might as mistake construct a path that has '//' in it. Technically, all programs will traverse that properly.


Also, we should try the pathlib '/' fancy :) https://docs.python.org/3/library/pathlib.html#operators